File: legacy_perl/merge_fastq.pl
#!/usr/bin/env perl
# Merge interleaved FASTQ or paired files (simple concatenation utility)
use strict;
use warnings;
my ($f1,$f2,$out) = @ARGV;
die "Usage: $0 r1.fastq r2.fastq out.fastq\n" unless $f1 && $f2 && $out;
open my $o, '>', $out or die $!;
open my $a, '<', $f1 or die $!;
open my $b, '<', $f2 or die $!;
while(<$a>){ print $o $_; $_ = <$b>; print $o $_ if defined $_; }
close $a; close $b; close $o;
print "Merged to $out\n";

File: biopython_rewrite/merge_fastq.py
#!/usr/bin/env python3
# Simple merge of two FASTQ files by interleaving lines
import sys
if len(sys.argv)!=4:
    print('Usage: merge_fastq.py r1.fastq r2.fastq out.fastq'); sys.exit(1)
r1, r2, out = sys.argv[1:4]
with open(r1) as a, open(r2) as b, open(out,'w') as o:
    while True:
        a_block = [a.readline() for _ in range(4)]
        b_block = [b.readline() for _ in range(4)]
        if not a_block[0] or not b_block[0]:
            break
        o.writelines(a_block); o.writelines(b_block)
print('Merged to', out)
